sopel/lifecycle.py

```diff
@@ -13,21 +13,21 @@
 import inspect
 import logging
 import traceback
-from typing import Callable, Optional
+from typing import Callable
```

Comment:
Same here: Is there a reason to prefer the deprecated `typing.Callable` over `collections.abc.Callable`?